shell/platform/darwin/ios/framework/Source/SemanticsObject.h

```diff
@@ -155,4 +155,50 @@ constexpr int32_t kRootNodeId = 0;
 - (instancetype)initWithSemanticsObject:(SemanticsObject*)semanticsObject;
 @end
 
+/**
+ * Represents a semantics object that has children and hence has to be presented to the OS as a
+ * UIAccessibilityContainer.
+ *
+ * The SemanticsObject class cannot implement the UIAccessibilityContainer protocol because an
+ * object that returns YES for isAccessibilityElement cannot also implement
+ * UIAccessibilityContainer.
+ *
+ * With the help of SemanticsObjectContainer, the hierarchy of semantic objects received from
+ * the framework, such as:
+ *
+ * SemanticsObject1
+ *     SemanticsObject2
+ *         SemanticsObject3
+ *         SemanticsObject4
+ *
+ * is translated into the following hierarchy, which is understood by iOS:
+ *
+ * SemanticsObjectContainer1
+ *     SemanticsObject1
+ *     SemanticsObjectContainer2
+ *         SemanticsObject2
+ *         SemanticsObject3
+ *         SemanticsObject4
+ *
+ * From Flutter's view of the world (the first tree seen above), we construct iOS's view of the
+ * world (second tree) as follows: We replace each SemanticsObjects that has children with a
+ * SemanticsObjectContainer, which has the original SemanticsObject and its children as children.
+ *
+ * SemanticsObjects have semantic information attached to them which is interpreted by
+ * VoiceOver (they return YES for isAccessibilityElement). The SemanticsObjectContainers are just
+ * there for structure and they don't provide any semantic information to VoiceOver (they return
+ * NO for isAccessibilityElement).
+ */
+@interface SemanticsObjectContainer : UIAccessibilityElement
```

Comment:
Any way we can start a policy of prefixing all classes with the `FLT` prefix (since this one has never been in a header before it's eligible even though it's not public)?  It's a poor namespacing substitute but it's all Obj-C's got.  We have random plugins using that prefix too, which is also not the right thing to do across frameworks...  The other classes at least start with `Flutter` which is still not good, but is at least some kind of prefix.
http://google.github.io/styleguide/objcguide.html#prefixes
https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/Conventions/Conventions.html#//apple_ref/doc/uid/TP40011210-CH10-SW4